controlplane/kubeadm/internal/filters.go

```diff
@@ -41,65 +40,73 @@ import (
 // - mutated in-place (ex: NodeDrainTimeout)
 // - are not dictated by KCP (ex: ProviderID)
 // - are not relevant for the rollout decision (ex: failureDomain).
-func matchesMachineSpec(infraConfigs map[string]*unstructured.Unstructured, machineConfigs map[string]*bootstrapv1.KubeadmConfig, kcp *controlplanev1.KubeadmControlPlane, machine *clusterv1.Machine) (string, bool, error) {
-	mismatchReasons := []string{}
+func matchesMachineSpec(infraConfigs map[string]*unstructured.Unstructured, machineConfigs map[string]*bootstrapv1.KubeadmConfig, kcp *controlplanev1.KubeadmControlPlane, machine *clusterv1.Machine) (bool, []string, []string, error) {
+	logReasons := []string{}
```

Comment:
Wondering if it would be easier to create a struct that contains both the log and the condition reason and then pass around slices of that struct.

This would mean we only have to pass around one slice and it feels a bit more typed

Reply:
i would prefer not to, mostly because this struct will have a semantic meaning when used in match* (mismatch reasons) func or in the need rollout func (rollout messages).
Also, outside of this package those values are used one at time